lib/ui5Framework/maven/Installer.js

```diff
@@ -0,0 +1,497 @@
+	 * is no longer required.
+	 */
+
+	/**
+	 * @param {object} parameters
+	 * @param {string} parameters.groupId GroupId of the requested artifact
+	 * @param {string} parameters.artifactId ArtifactId of the requested artifact
+	 * @param {string} parameters.version Version of the requested artifact
+	 * @param {string} parameters.classifier Classifier of the requested artifact
+	 * @param {string} parameters.extension Extension of the requested artifact
+	 * @param {string} [parameters.revision] Optional revision of the artifact to request.
+	 * 	If not provided the latest revision will be determined via the registry metadata.
+	 * @returns {@ui5/project/ui5Framework/maven/Installer~InstalledArtifact}
+	 *
+	 */
+	async installArtifact({groupId, artifactId, version, classifier, extension, revision}) {
+		if (!revision) {
+			const metadata = await this._fetchArtifactMetadata({
+				groupId, artifactId, version, classifier, extension
+			});
+			revision = metadata.revision;
+		}
+		const coordinates = {
+			groupId, artifactId,
+			version, revision,
+			classifier, extension
+		};
+
+		const targetPath = this._getTargetPathForArtifact(coordinates);
+		const installed = await this._pathExists(targetPath);
+		const logId = this._generateLogIdFromCoordinates(coordinates);
+		const fsId = this._generateFsIdFromCoordinates(coordinates);
+		if (!installed) {
+			await this._synchronize(`artifact-${fsId}`, async () => {
+				// check again whether the artifact is now installed
+				const installed = await this._pathExists(targetPath);
+				if (installed) {
+					log.verbose(`Already installed: ${artifactId} in version ${revision}`);
+					return;
+				}
+
+				const stagingPath = this._getStagingPathForArtifact(coordinates);
+				log.info(`Installing missing artifact ${logId}...`);
+
+				// Check whether staging dir already exists and remove it
+				if (await this._pathExists(stagingPath)) {
+					log.verbose(`Removing existing file in staging dir at ${stagingPath}...`);
+					await rm(stagingPath);
+				}
+				await mkdirp(path.dirname(stagingPath));
+
+				log.verbose(`Installing ${artifactId} in version ${version} to ${stagingPath}...`);
+
+				// TODO: Stream response body to installPackage and unzip directly via
+				// https://github.com/isaacs/minizlib (already in dependencies through pacote)
+				// This way we do not store the archive unnecessarily
+				await this.getRegistry().requestArtifact(coordinates, stagingPath);
+
+				await mkdirp(path.dirname(targetPath));
+				log.verbose(
+					`Promoting artifact from staging path ${stagingPath} to target path at ${targetPath}...`);
+				await rename(stagingPath, targetPath);
+			});
+		} else {
+			log.verbose(`Already installed: ${artifactId} in version ${revision}`);
+		}
+		return {
+			artifactPath: targetPath,
+			removeArtifact: () => {
+				return rm(targetPath);
+			}
+		};
+	}
+
+	async _projectExists(targetDir) {
```

Comment:
in general we can use private functions with `#` here instead of putting only the `_` in the function name.

Reply:
Completelly agree with that, but there are some considerations we need to take into account.

The other Installer has been already made that way. Current implementation would become a lot bigger if we do such reffactoring here. Also we'd need to adjust lots of tests. Maybe we could consider such a refactoriing later.

Reply:
would be fine for me.